### ISSUE TITLE
[Bug]: Low render distance for dragons

### ISSUE THREAD

**Issue by Goldenshadow86** [Alex-the-666/Ice_and_Fire#5194]
### What happened?

For some reason dragons on my server have a really low render distance. To the point where we are dying when its like 30+ blocks out of it's render distance. I have tried to change all my render setting on the server side and client side but nothing seems to work.

![Medal_246EL89yls](https://github.com/AlexModGuy/Ice_and_Fire/assets/29588612/5a1e8ca1-e49d-48f6-b0e2-9239dff795bf)
![Medal_hMhaTNVToH](https://github.com/AlexModGuy/Ice_and_Fire/assets/29588612/f0259a80-9f8e-4281-9fb3-1c909e160b88)
[2024-02-24-2.log](https://github.com/AlexModGuy/Ice_and_Fire/files/14394325/2024-02-24-2.log)
[debug-2.log](https://github.com/AlexModGuy/Ice_and_Fire/files/14394326/debug-2.log)

### Minecraft Version

1.20.1

### Ice and Fire version

iceandfire-2.1.13-1.20.1-beta-2.jar

### Citadel version

citadel-2.4.9-1.20.1.jar

### Other Information

- [X] I am using other mods besides Ice and Fire
- [X] I have attached a log file from the session that I have experienced this issue

**Comment by SiverDX** [comment]
isn't the range determined by `view-distance` of `server.properties`?

**Comment by Goldenshadow86** [comment]
> isn't the range determined by `view-distance` of `server.properties`?

I tried to change it all the way up to 20 and it did nothing

**Comment by SiverDX** [comment]
does this happen without NoSeeNoTick

**Comment by SiverDX** [comment]
> > isn't the range determined by `view-distance` of `server.properties`?
> 
> I tried to change it all the way up to 20 and it did nothing

also regarding this
you changed config, restarted server and made sure that your client render/view distance is also higher?

**Comment by Goldenshadow86** [comment]
> > > isn't the range determined by `view-distance` of `server.properties`?
> > 
> > 
> > I tried to change it all the way up to 20 and it did nothing
> 
> also regarding this you changed config, restarted server and made sure that your client render/view distance is also higher?

Yes after every change I restarted the server. Also I dont know what "NoSeeNoTick" is

**Comment by SiverDX** [comment]
> > > > isn't the range determined by `view-distance` of `server.properties`?
> > > 
> > > 
> > > I tried to change it all the way up to 20 and it did nothing
> > 
> > 
> > also regarding this you changed config, restarted server and made sure that your client render/view distance is also higher?
> 
> Yes after every change I restarted the server. Also I dont know what "NoSeeNoTick" is

a mod you have installed https://www.curseforge.com/minecraft/mc-mods/no-see-no-tick

**Comment by Goldenshadow86** [comment]
ahh okay I just saw it in the mod pack. (I'm using Prominence II) Should I try to remove it or is there a config I can mess with

> > > > > isn't the range determined by `view-distance` of `server.properties`?
> > > > 
> > > > 
> > > > I tried to change it all the way up to 20 and it did nothing
> > > 
> > > 
> > > also regarding this you changed config, restarted server and made sure that your client render/view distance is also higher?
> > 
> > 
> > Yes after every change I restarted the server. Also I dont know what "NoSeeNoTick" is
> 
> a mod you have installed https://www.curseforge.com/minecraft/mc-mods/no-see-no-tick

**Comment by Goldenshadow86** [comment]
> > > > > isn't the range determined by `view-distance` of `server.properties`?
> > > > 
> > > > 
> > > > I tried to change it all the way up to 20 and it did nothing
> > > 
> > > 
> > > also regarding this you changed config, restarted server and made sure that your client render/view distance is also higher?
> > 
> > 
> > Yes after every change I restarted the server. Also I dont know what "NoSeeNoTick" is
> 
> a mod you have installed https://www.curseforge.com/minecraft/mc-mods/no-see-no-tick

I tried to edit the config and tried removing it did not work. (I restarted the server each time I made an edit)

**Comment by SiverDX** [comment]
are other entities rendered the same way / limited to the same distance

**Comment by Goldenshadow86** [comment]
yes they are I spawned a zombie and it goes away at the same time

**Comment by SiverDX** [comment]
might be more helpful to ask the modpack maker what kind of (performance) mods or default configuration they have which could cause this

**Comment by Alccardi** [comment]
Had the same problem, here’s the solution to this:

You need to edit the **noseenotick-common.toml** configuration file on both:
- **The server** (located in `/config`)
- **Every player’s client** (located in `.minecraft/config`)

The main settings you need to change to fix this issue are:

- `maxEntitySpawnDistanceHorizontal` – increases how far away entities like dragons can spawn.
- `maxEntityTickDistanceHorizontal` – increases how far away entities like dragons can move, attack, and animate.
- `entityIgnoreList` – adds a list of important entities (like Ice and Fire dragons) that should **always** be active, regardless of distance.

Below is my configuration with comments explaining what each setting does:
<pre>

# ==============================
# NoSeeNoTick - Recommended Config for Servers with Ice and Fire Dragons
# NoSeeNoTick helps improve server performance by limiting how far away entities spawn, tick, and render.
# This is great for normal mobs, but breaks large, fast-moving, or important entities like dragons.
# This config raises those limits and adds a whitelist to protect important mobs.

[general]
	# Maximum Spawn Distances
	# These control how far from players mobs can spawn.
	# Distances are in blocks squared (so 50176 = 224 blocks horizontal radius).
	# Default was 64^2 (too low for dragons).
	maxEntitySpawnDistanceHorizontal = 50176  # 224 blocks / 14 chunks
	maxEntitySpawnDistanceVertical = 128      # 128 blocks / 8 chunks (up/down)

	# Maximum Ticking Distances
	# These control how far from players mobs can "tick" (move, attack, etc.).
	# Distances are in blocks squared (so 50176 = 224 blocks horizontal radius).
	maxEntityTickDistanceHorizontal = 50176   # 224 blocks / 14 chunks
	maxEntityTickDistanceVertical = 128       # 128 blocks / 8 chunks

	# Entity Whitelist (Ignore List)
	# Entities on this list will ALWAYS spawn, tick, and render at full distances.
	# Use either individual entities like "minecraft:creeper"
	# or wildcard for entire mods like "iceandfire:*"
	# This list includes all Ice and Fire entities and common modded machines.
	entityIgnoreList = [
		"minecraft:wither",                  # Vanilla boss
		"minecraft:phantom",                 # Flying mob
		"minecraft:ender_dragon",            # Vanilla Ender Dragon
		"minecraft:elder_guardian",          # Ocean boss
		"minecraft:player",                  # Players always tick
		"alexsmobs:void_worm",                # Boss from Alex's Mobs
		"alexsmobs:void_worm_part",           # Void worm parts
		"alexsmobs:void_worm_shot",           # Void worm projectiles
		"minecraft:enderman",                 # Common hostile
		"minecraft:creeper",                  # Explosive classic
		"minecraft:skeleton",                 # Ranged classic
		"minecraft:zombie",                   # Melee classic
		"minecraft:zombie_villager",          # Villager conversion
		"iceandfire:*",                        # All Ice and Fire mobs (dragons, etc.)
		"create:*",                           # All Create mod entities
		"botania:*",                          # All Botania entities
		"ftbic:*",                            # All FTB Industrial Contraptions
		"immersiveengineering:*",             # Immersive Engineering
		"ae2:*",                              # Applied Energistics 2
		"littlelogistics:*",                  # Little Logistics
		"tiab:*"                              # Time In A Bottle entities
	]

	# Minimum Players Requirement
	# NoSeeNoTick only activates when at least this many players are online.
	# Set to 1 if you want this enabled in singleplayer.
	# Set to 2 if you only want this for multiplayer performance.
	minPlayers = 2

	# Notes on Distances and Performance
	# - 1 chunk = 16 blocks.
	# - 224 blocks = 14 chunks.
	# - Vanilla Minecraft usually ticks entities within 10 chunks (~160 blocks).
	# - These values extend that to 14 chunks (224 blocks), ensuring dragons stay active even at distance.
	#
	# This change slightly increases server load but prevents:
	# - Dragons disappearing mid-fight.
	# - Invisible fireballs/projectiles.
	# - Unfair dragon fights where dragons freeze but keep attacking.

</pre>
